File: app/src/main/java/com/example/adam/myapplication/data/TaskDatabase.java
package com.example.adam.myapplication.data;

import android.arch.persistence.room.Database;
import android.arch.persistence.room.Room;
import android.arch.persistence.room.RoomDatabase;
import android.arch.persistence.room.TypeConverters;
import android.content.Context;

@Database(entities = Task.class, version = 1)
@TypeConverters(Converters.class)
public abstract class TaskDatabase extends RoomDatabase {

    private static final String DATABASE_NAME = "tasks_db";
    private static TaskDatabase INSTANCE;

    public abstract TaskDao getTaskDao();

    public static TaskDatabase getInstance(Context context) {
        if (INSTANCE == null) {
            INSTANCE = Room.databaseBuilder(context, TaskDatabase.class, DATABASE_NAME).build();
        }

        return INSTANCE;
    }
}

File: app/src/main/java/com/example/adam/myapplication/data/db/task/TaskDatabase.kt
package com.example.adam.myapplication.data.db.task

import android.arch.persistence.room.Database
import android.arch.persistence.room.Room
import android.arch.persistence.room.RoomDatabase
import android.arch.persistence.room.TypeConverters
import android.content.Context

import com.example.adam.myapplication.data.objects.Task

@Database(entities = [Task::class], version = 1)
@TypeConverters(Converters::class)
abstract class TaskDatabase : RoomDatabase() {

    abstract val taskDao: TaskDao

    companion object {

        private const val DATABASE_NAME = "tasks_db"
        private var INSTANCE: TaskDatabase? = null

        fun getInstance(context: Context): TaskDatabase {
            return INSTANCE
                    ?: Room.databaseBuilder(context, TaskDatabase::class.java, DATABASE_NAME).build()
        }
    }
}